pkg/builder/project.go

```diff
@@ -191,3 +194,14 @@ func injectDependencies(ctx *builderContext) error {
 func sanitizeDependencies(ctx *builderContext) error {
 	return camel.SanitizeIntegrationDependencies(ctx.Maven.Project.Dependencies)
 }
+
+func injectProfile(ctx *builderContext) error {
```

Comment:
Instead of injecting the profile (which could be seen as dangerous against the immutability of the project provided) I wonder if it would be possible to have instead the profile called in the `./mvnw package -P custom-profile.xml` being the custom-profile.xml created from the configmap/secret (as I think we do the same for the settings.xml).

Reply:
I analyzed this option. There is a way to use an external file but it will limit the build functionnalities overides from the profile as stated in [the maven documentation](https://maven.apache.org/guides/introduction/introduction-to-profiles.html#which-areas-of-a-pom-can-be-customized-by-each-type-of-profile-w). This is the reason I decided to inject the content into the pom file. Let me know what you think.

Reply:
I see. Thanks for clarifying. Then, it's good to go for me. Please, wait the fix I'm doing for https://github.com/apache/camel-k/issues/4597 before rebasing.